### ISSUE TITLE
[TypeScript-Resolvers] Default arguments in queries/mutations are generated as Maybe

### ISSUE THREAD

**Issue by RIP21** [dotansimha/graphql-code-generator#2064]
**Describe the bug**
If you have default arguments on SDL side.
```
limit: PositiveInt = 100
offset: NonNegativeInt = 0
```
It will generate:
```
limit?: Maybe<Scalar['PositiveInt']>
offset?: Maybe<Scalar['NonNegativeInt']>
```

**Expected behavior**
Should be generated:
```
limit: Scalar['PositiveInt']
offset: Scalar['NonNegativeInt']
```

**Environment:**
codegen: 1.3.1
node: v12.3.1

**Comment by RIP21** [comment]
The similar issue back in the days for Flow.
https://github.com/dotansimha/graphql-code-generator/issues/1097

**Comment by dotansimha** [comment]
It depends, because for `typescript-operations`, that's fine, because it means that you don't have to specify this field, because it has a default value. 
It might be an issue only with `typescript-resolvers`, because there if there is a default value, it means that that the field *must* be there.

@kamilkisiela what do you think? maybe we can keep it as-is now (with `Maybe`), and do something like `UnMaybe<MyQueryArgs, 'fieldA'>` in the usage of the resolvers plugins?

**Comment by dotansimha** [comment]
Fixed in: https://github.com/dotansimha/graphql-code-generator/pull/2122 
@RIP21 can you please try it with `1.3.1-alpha-5d4a4dba.80`?

**Comment by yamafaktory** [comment]
@dotansimha I'm facing the same issue with the latest alpha.

This schema:
```graphql
"Query for search results"
  search(
    "The search text"
    input: SearchInput!

    "The page size"
    first: Int = 20

    "Page offset id"
    after: ID = "0"

    "Page number, if not using the 'after' parameter"
    page: Int
  ): SearchResultsConnection!
```

Is turned into:
```typescript
/** Root query field */
export type QuerySearchArgs = {
  input: SearchInput
  first?: Maybe<Scalars['Int']>
  after?: Maybe<Scalars['ID']>
  page?: Maybe<Scalars['Int']>
}
```
Instead of:
```typescript
/** Root query field */
export type QuerySearchArgs = {
  input: SearchInput
  first: Scalars['Int']
  after: Scalars['ID']
  page?: Maybe<Scalars['Int']>
}
```
Thanks for your work 👍.

**Comment by dotansimha** [comment]
@yamafaktory The `Args` type should look the same as before, but we are removing the `?` and `Maybe` modifiers when it's used in the resolvers signature. 
Try to use the `QueryResolvers` and implement a resolver, and these fields will appear as non-optional.

**Comment by yamafaktory** [comment]
@dotansimha thanks a lot for the feedback! Somehow I would have expected `QuerySearchArgs` not to have any `Maybe` in my case.

**Comment by dotansimha** [comment]
Fixed in 1.4.0 🎉
